internal/goldendataset/metric_gen.go

```diff
@@ -0,0 +1,201 @@
+	ilms := rm.InstrumentationLibraryMetrics()
+	ilms.Resize(cfg.NumIlm)
+	for i := 0; i < cfg.NumIlm; i++ {
+		ilm := ilms.At(i)
+		populateMetrics(cfg, ilm)
+	}
+}
+
+func populateMetrics(cfg MetricCfg, ilm pdata.InstrumentationLibraryMetrics) {
+	metrics := ilm.Metrics()
+	metrics.Resize(cfg.NumMetrics)
+	for i := 0; i < cfg.NumMetrics; i++ {
+		metric := metrics.At(i)
+		metric.InitEmpty()
+		populateMetricDesc(cfg, metric)
+		switch cfg.MetricDescriptorType {
+		case pdata.MetricTypeInt64, pdata.MetricTypeMonotonicInt64:
+			populateIntPoints(cfg, metric)
+		case pdata.MetricTypeDouble, pdata.MetricTypeMonotonicDouble:
+			populateDblPoints(cfg, metric)
+		case pdata.MetricTypeHistogram:
+			populateHistogramPoints(cfg, metric)
+		case pdata.MetricTypeSummary:
+			populateSummaryPoints(cfg, metric)
+		}
+	}
+}
+
+func populateMetricDesc(cfg MetricCfg, metric pdata.Metric) {
+	desc := metric.MetricDescriptor()
+	desc.InitEmpty()
+	desc.SetName("my-md-name")
+	desc.SetDescription("my-md-description")
+	desc.SetUnit("my-md-units")
+	desc.SetType(cfg.MetricDescriptorType)
+}
+
+func populateIntPoints(cfg MetricCfg, metric pdata.Metric) {
+	pts := metric.Int64DataPoints()
+	pts.Resize(cfg.NumPts)
+	for i := 0; i < cfg.NumPts; i++ {
+		pt := pts.At(i)
+		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
+		pt.SetValue(cfg.PtVal + int64(i))
+		populatePtLabels(cfg, pt.LabelsMap())
+	}
+}
+
+func populateDblPoints(cfg MetricCfg, metric pdata.Metric) {
+	pts := metric.DoubleDataPoints()
+	pts.Resize(cfg.NumPts)
+	for i := 0; i < cfg.NumPts; i++ {
+		v := cfg.PtVal + int64(i)
+		pt := pts.At(i)
+		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
+		pt.SetValue(float64(v))
+		populatePtLabels(cfg, pt.LabelsMap())
+	}
+}
+
+func populateHistogramPoints(cfg MetricCfg, metric pdata.Metric) {
+	pts := metric.HistogramDataPoints()
+	pts.Resize(cfg.NumPts)
+	for i := 0; i < cfg.NumPts; i++ {
+		pt := pts.At(i)
+		pt.SetStartTime(pdata.TimestampUnixNano(cfg.StartTime))
+		pt.SetTimestamp(getTimestamp(cfg.StartTime, cfg.StepSize, i))
+		populatePtLabels(cfg, pt.LabelsMap())
+		setHistogramBounds(pt, 1, 2, 3, 4, 5)
+		addHistogramVal(pt, 1)
+		addHistogramVal(pt, 3)
+		addHistogramVal(pt, 3)
+		addHistogramVal(pt, 5)
+	}
+}
+
+func setHistogramBounds(hdp pdata.HistogramDataPoint, bounds ...float64) {
+	hdp.Buckets().Resize(len(bounds))
+	hdp.SetExplicitBounds(bounds)
+}
+
+func addHistogramVal(hdp pdata.HistogramDataPoint, val float64) {
+	hdp.SetCount(hdp.Count() + 1)
+	hdp.SetSum(hdp.Sum() + val)
+	buckets := hdp.Buckets()
+	bounds := hdp.ExplicitBounds()
+	for i := 0; i < len(bounds); i++ {
+		bound := bounds[i]
+		if val <= bound {
+			bucket := buckets.At(i)
+			bucket.SetCount(bucket.Count() + 1)
+			break
+		}
+	}
+}
+
+func populateSummaryPoints(cfg MetricCfg, metric pdata.Metric) {
```

Comment:
May want to add quantiles/percentiles as well

Reply:
Done